### PR TITLE
upgrade to junit 5 

### PR DESCRIPTION
**Pull request by piomin**
In  the newest version some tests based on JUnit4 does not work. I fixed it by adding JUnit5 support for spring-cloud-kubernetes-leader module, and by adding junit-vintage-engine to spring-cloud-kubernetes-examples module since Arquillian Cube does not support JUnit5

### CODE REVIEW

File: spring-cloud-kubernetes-examples/kubernetes-leader-election-example/pom.xml [hunk 1]
```diff
@@ -78,6 +78,14 @@
 					<skip>true</skip>
 				</configuration>
 			</plugin>
+			<plugin>
+				<artifactId>maven-surefire-plugin</artifactId>
+				<version>2.22.2</version>
```
Comment (spencergibb) [inline spring-cloud-kubernetes-examples/kubernetes-leader-election-example/pom.xml:83]: This should happen in spring-cloud-build

Comment (piomin) [reply spring-cloud-kubernetes-examples/kubernetes-leader-election-example/pom.xml:83]: I see that currently these version is taken from parent pom.xml (branch master). I changed both versions for maven-failsafe-plugin and maven-surefire-plugin in the parent pom.xml.

		<plugins>
			<plugin>
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-failsafe-plugin</artifactId>
				<version>${maven-failsafe-plugin.version}</version>
			</plugin>
		</plugins>

I don't know - maybe it should be taken from spring-cloud-build, but I'm guessing that it is the change for the whole Spring Cloud project. I had to set version of maven-surefire-plugin to 2.22.2 to make test works after changing enabling junit-jupiter for spring-cloud-kubernetes-leader module.

**Comment by ryanjbaxter** [comment]
All tests should be passing and have been passing on CircleCI https://app.circleci.com/pipelines/github/spring-cloud/spring-cloud-kubernetes

**Comment by piomin** [comment]
Well, I just replaced junit-vintage with junit-jupiter here. The tests were passing...

**Comment by spencergibb** [comment]
yeah, this should be upgrade to junit 5 now instead of fixing errors.

**Comment by spencergibb** [comment]
you'll need to rebase

**Comment by piomin** [comment]
@spencergibb  Is it ok now?

File: spring-cloud-kubernetes-leader/src/test/java/org/springframework/cloud/kubernetes/leader/LeaderAutoConfigurationTests.java [hunk 2]
```diff
@@ -16,21 +16,21 @@
...
 import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.http.MediaType;
-import org.springframework.test.context.junit4.SpringRunner;
 import org.springframework.test.web.reactive.server.WebTestClient;
 
 import static org.hamcrest.Matchers.containsString;
 
-@RunWith(SpringRunner.class)
+@ExtendWith(MockitoExtension.class)
```
Comment (Haybu) [inline spring-cloud-kubernetes-leader/src/test/java/org/springframework/cloud/kubernetes/leader/LeaderAutoConfigurationTests.java:33]: @piomin: won't this be a good case to use Spring TextContext leveraging @ExtendWith(SpringExtension)?

Comment (Haybu) [reply spring-cloud-kubernetes-leader/src/test/java/org/springframework/cloud/kubernetes/leader/LeaderAutoConfigurationTests.java:33]: could be changed later, not breaking the test for now

**Pull request merged**